src/main/java/org/peergos/LogExporter.java

```diff
@@ -0,0 +1,167 @@
+package org.peergos;
+
+import com.google.gson.Gson;
+
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.util.EntityUtils;
+import org.peergos.util.Logging;
+
+import java.util.logging.Logger;
+import java.io.IOException;
+import java.io.BufferedInputStream;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileWriter;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.logging.Level;
+
+public class LogExporter {
+    private static final Logger LOG = Logging.LOG();
+    private static final Gson gson = new Gson();
+    private static final int SLEEP_TIME_MILLISECONDS = 10_000;
+    private static final String LOG_FILE_PATH = Nabu.DEFAULT_IPFS_DIR_PATH.toAbsolutePath().toString() + "/trace.log";
+    private static final String STATE_PATH = Nabu.DEFAULT_IPFS_DIR_PATH.toAbsolutePath().toString()
+            + "/log-exporter-state.log";
+
+    private String exportEndpoint;
+    private ZonedDateTime lastWritten;
+    private DateTimeFormatter timestampFormatter;
+
+    private FileInputStream inFile = null;
+    private BufferedInputStream buffer = null;
+
+    private final CloseableHttpClient httpClient = HttpClients.createDefault();
+
+    public LogExporter(String exportEndpoint) throws Exception {
+        this.exportEndpoint = exportEndpoint;
+        // TODO(@millerm) - read from state upon initialization
+        this.lastWritten = ZonedDateTime.ofInstant(java.time.Instant.EPOCH, ZoneOffset.UTC);
+        this.timestampFormatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss 'UTC'");
+    }
+
+    public void writeState() {
+        try (BufferedWriter writer = new BufferedWriter(new FileWriter(STATE_PATH))) {
+            writer.write(this.lastWritten.format(this.timestampFormatter));
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+        System.out.println("Wrote state.");
+    }
+
+    public Map<String, Object> pollNewLogs() {
+        System.out.println("Polling new logs...");
+        Map<String, Object> logRecord = new HashMap<>();
+
+        try {
+            inFile = new FileInputStream(LOG_FILE_PATH);
+            buffer = new BufferedInputStream(inFile);
+
+            int rawLogData;
+
+            // Handle possible resource contention b/t writer & reader
+            while ((rawLogData = buffer.read()) != -1) {
+                System.out.print((char) rawLogData);
+            }
```

Comment:
I think we will need to actually use an OS lock on this file from both the reader and writer since they are running on different processes / JVMs. Will update. 

See: https://docs.oracle.com/javase/8/docs/api/java/nio/channels/FileLock.html